x-pack/plugins/lens/public/datasources/text_based/components/dimension_editor.tsx

```diff
@@ -122,6 +174,43 @@ export function TextBasedDimensionEditor(props: TextBasedDimensionEditorProps) {
           {props.dataSectionExtra}
         </div>
       )}
+      {!isFullscreen && (
+        <div className="lnsIndexPatternDimensionEditor--padded lnsIndexPatternDimensionEditor--collapseNext">
+          {selectedField && (
```

Comment:
nit: you can bring `selectedField` up to the initial condition and remove it from the rest of this code.
```suggestion
      {!isFullscreen && selectedField && (
        <div className="lnsIndexPatternDimensionEditor--padded lnsIndexPatternDimensionEditor--collapseNext">

```